src/obj-pile.c

```diff
@@ -798,12 +805,10 @@ struct object *object_split(struct object *src, int amt)
 	/* Modify quantity */
 	dest->number = amt;
 	src->number -= amt;
-	if (src->note)
-		dest->note = src->note;
 	if (src->known) {
 		dest->known->number = dest->number;
 		src->known->number = src->number;
-		dest->known->note = src->known->note;
+		dest->known->note = string_make(src->known->note);
```

Comment:
Since dest->known was copied from src->known, dest->know->note is already set appropriately and setting it again here will leak memory.